tests/test_middleware_interface.py

```diff
@@ -76,8 +75,22 @@ def fake_file_data(filename, dimensions, instrument):
 class MiddlewareInterfaceTest(unittest.TestCase):
     """Test the MiddlewareInterface class with faked data.
     """
+    @classmethod
+    def setUpClass(cls):
+        cls.env_patcher = unittest.mock.patch.dict(os.environ,
+                                                   {"IP_APDB": "localhost"})
+        cls.env_patcher.start()
+
+        super().setUpClass()
+
+    @classmethod
+    def tearDownClass(cls):
+        super().tearDownClass()
+
+        cls.env_patcher.stop()
 
     def setUp(self):
+        from activator.middleware_interface import MiddlewareInterface
```

Comment:
Why does this need to go here?

Reply:
No idea. This is from what Eric pushed. Removed.

Reply:
Ah, it was because of the global `os.environ["IP_APDB"]` in middleware_interface. I've made that a class member now, so it doesn't need to be defined at import.